### ISSUE TITLE
transient SIGSEGV using btrfs sub find

### ISSUE THREAD

**Issue by cmurf** [kdave/btrfs-progs#300]
Trying to find new files being created and see two crashes over about 25 minutes. So, maybe 1 in 10 times it crashes?

kernel-5.8.9-301.fc33.x86_64
btrfs-progs-5.7-5.fc33.x86_64
```
Wed 2020-09-16 01:28:03 MDT  120174     0     0  11 present   /usr/sbin/btrfs
# coredumpctl gdb 120174
           PID: 120174 (btrfs)
           UID: 0 (root)
           GID: 0 (root)
        Signal: 11 (SEGV)
     Timestamp: Wed 2020-09-16 01:28:03 MDT (3min 30s ago)
  Command Line: btrfs sub find / 39540
    Executable: /usr/sbin/btrfs
 Control Group: /user.slice/user-1000.slice/session-15.scope
          Unit: session-15.scope
         Slice: user-1000.slice
       Session: 15
     Owner UID: 1000 (chris)
       Boot ID: 13ad8baf3bf148b2b21104acf13f363f
    Machine ID: b51b4a725db84fd286dcf4a790a50a1d
      Hostname: fmac.local
       Storage: /var/lib/systemd/coredump/core.btrfs.0.13ad8baf3bf148b2b21104acf13f363f.120174.1600241283000000.zst
       Message: Process 120174 (btrfs) of user 0 dumped core.
                
                Stack trace of thread 120174:
                #0  0x0000564c6381b88d btrfs_list_find_updated_files (btrfs + 0x6288d)
                #1  0x0000564c637e171b cmd_subvol_find_new.lto_priv.0 (btrfs + 0x2871b)
                #2  0x0000564c637cedd0 main (btrfs + 0x15dd0)
                #3  0x00007fd74aee01a2 __libc_start_main (libc.so.6 + 0x281a2)
                #4  0x0000564c637cf10e _start (btrfs + 0x1610e)

GNU gdb (GDB) Fedora 9.2-7.fc33
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-redhat-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /usr/sbin/btrfs...
Reading symbols from /usr/lib/debug/usr/sbin/btrfs-5.7-5.fc33.x86_64.debug...
[New LWP 120174]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Core was generated by `btrfs sub find / 39540'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  memcpy (__len=<optimized out>, __src=<optimized out>, __dest=<optimized out>) at btrfs-list.c:1766
1766				if (sh.len == 0)
(gdb) thread apply all bt full

Thread 1 (Thread 0x7fd74aeb58c0 (LWP 120174)):
#0  memcpy (__len=<optimized out>, __src=<optimized out>, __dest=<optimized out>) at btrfs-list.c:1766
No locals.
#1  btrfs_list_find_updated_files (fd=3, root_id=<optimized out>, oldest_gen=39540) at btrfs-list.c:1759
        ret = 0
        args = {key = {tree_id = 0, min_objectid = 1, max_objectid = 18446744073709551615, min_offset = 10123776, max_offset = 18446744073709551615, min_transid = 39540, max_transid = 18446744073709551615, min_type = 218103808, max_type = 108, nr_items = 43, unused = 0, unused1 = 0, unused2 = 0, unused3 = 0, unused4 = 0}, buf = "\000\232\000\000\000\000\000\000 \247\b\000\000\000\000\000\b\034{H\000\000\000\000T\000\000\000.\000\000\000+\247\b\000\000\000\000\000\001\000\000\000\000\000\000\000\000z\232\000\000\000\000\000\000\000\000\020\000\001pmbus_core.ko.xzz\232\000\000\000\000\000\000 \247\b\000\000\000\000\000\271a\334I\000\000\000\000T\000\000\000,\000\000\000(\247\b\000\000\000\000\000\001\000\000\000\000\000\000\000\000z\232\000\000\000\000\000\000\000\000\016\000\001max34440.ko.xzz\232\000\000\000\000\000\000 \247\b\000\000\000\000\000n\326=\\\000\000\000\000T\000\000\000+\000\000\000#\247\b\000\000\000\000\000\001\000\000\000\000\000\000\000\000"...}
        sk = 0x7ffc04c85dc0
        sh = {transid = 567087, objectid = 1, offset = 10123776, type = 218103808, len = 1668612352}
        item = <optimized out>
        off = 1668614259
        found_gen = <optimized out>
        max_found = <optimized out>
        i = 26
        cache_dirid = 566958
        cache_ino = 567071
        cache_dir_name = 0x564c651a73e0 "usr/lib/modules/5.9.0-0.rc5.11.fc34.x86_64+debug/kernel/drivers/hwmon/"
        cache_full_name = 0x564c651a75a0 "usr/lib/modules/5.9.0-0.rc5.11.fc34.x86_64+debug/kernel/drivers/hwmon/pcf8591.ko.xz"
        backup = {generation = 0, ram_bytes = 0, compression = 0 '\000', encryption = 0 '\000', other_encoding = 0, type = 0 '\000', disk_bytenr = 0, disk_num_bytes = 0, offset = 0, num_bytes = 0}
#2  0x0000564c637e171b in cmd_subvol_find_new (cmd=<optimized out>, argc=<optimized out>, argv=<optimized out>) at cmds/subvolume.c:972
        fd = 3
        ret = <optimized out>
        subvol = <optimized out>
        last_gen = 39540
        dirstream = 0x564c6519e2a0
        err = BTRFS_UTIL_OK
#3  0x0000564c637cedd0 in cmd_execute (argv=0x7ffc04c88f90, argc=4, cmd=0x564c638aa5c0 <cmd_struct_subvolume>) at cmds/commands.h:125
No locals.
#4  main (argc=4, argv=0x7ffc04c88f90) at btrfs.c:402
        cmd = 0x564c638aa5c0 <cmd_struct_subvolume>
        bname = <optimized out>
        ret = <optimized out>
(gdb) quit
# 
```

**Comment by Damenly** [comment]
Succeeded to reproduce the issue in v5.9-rc4 and it was already fixed in v5.9 by the commit
https://lore.kernel.org/linux-btrfs/78bf853ef63f2dbd62587bb02bb723fbaa77c198.1600070418.git.fdmanana@suse.com/.

Issue #303 should be fixed too. Both of them happened after a tree search ioctl call.

**Comment by kdave** [comment]
Fixed in kernel, thanks for the testing.
